### PR TITLE
Mlp pytorch

### LINKED ISSUE #8: [PyTorch] Implementar a MLP
### Objetivo
Criar a rede neural principal do projeto em PyTorch.

### Descrição
Definir a arquitetura da MLP, função de ativação, função de perda e otimizador.
Garantir que a entrada da rede esteja compatível com o pipeline de features.
Deixar a arquitetura parametrizável para experimentação.

### Checklist
- [x] Definir arquitetura da MLP
- [x] Definir função de ativação
- [x] Definir loss function
- [x] Definir optimizer
- [x] Tornar hiperparâmetros configuráveis

### PR DESCRIPTION
**Pull request by fabiofonv**
Criei a mlp e adicionando o treino ao makefile

### CODE REVIEW

File: src/pytorch_mlp.py [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+            nn.Dropout(dropout_rate),
+            nn.Linear(hidden_dim // 2, 1)  # Saída única para classificação binária
+        )
+
+    def forward(self, x):
+        # A saída bruta (logits). A ativação Sigmoid será aplicada pela Loss Function (BCEWithLogitsLoss)
+        return self.network(x)
+
+
+def run_training(
```
Comment (crisleymarques) [inline src/pytorch_mlp.py]: para deixar mais legível sugiro dividir essa função que dá para reduzir a quantidade de comentários e ficar melhor separado tbm.

poderia ser algo assim:

prepare_data(batch_size)
train_model(...)
evaluate_model(...)


def run_training(...)

Comment (fabiofonv) [reply src/pytorch_mlp.py]: Certo, como você fez o train_baseline, certo?

Comment (fabiofonv) [reply src/pytorch_mlp.py]: Pronto, refatorado. Você acha que devia mudar o nome do arquivo pra train_mlp.py pra ficar um padrão?

Comment (crisleymarques) [reply src/pytorch_mlp.py]: sim

Comment (crisleymarques) [reply src/pytorch_mlp.py]: fiz aqui [e348604](https://github.com/crisleymarques/postech-ml-challenge-fase-1/pull/28/commits/e34860431bdb6d30105ee0d603bb5beb3ecfddb4)

**Review by crisleymarques — APPROVED** [review]
